docs/conceptual/runtime/contracts/ink.md

```diff
@@ -0,0 +1,108 @@
+---
+title: ink!
+---
+
+ink! is a [Rust](https://www.rust-lang.org/)-based embedded domain specific language
+([eDSL](https://wiki.haskell.org/Embedded_domain_specific_language)) for writing
+[Wasm](https://webassembly.org/) smart contracts specifically for the [SRML Contracts module](conceptual/runtime/contracts/contracts_module.md).
+
+## Design
+
+The main goals of ink! are user friendliness, conciseness, and efficiency.
+
+ink! is designed to be as close to the Rust programming language as possible. The language uses
+[attribute macros](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) to
+tag standard Rust structures into understandable contract components.
+
+```rust
+#[ink(...)]
+```
+
+Because ink! follows Rust standards, tools like `rustfmt` and `rust-analyzer` already work out of
+the box.
+
+### Contract Components
+
+ink! should still feel familiar to developers who have programmed using other modern smart contract
+languages. The skeleton of a contract has all of the same components that you might expect:
+
+  * Events - `#[ink(event)]`
+    * Event Topics - `#[ink(topic)]`
+  * Storage - `#[ink(storage)]`
+  * Constructor Functions - `#[ink(constructor)]`
+  * Message Functions - `#[ink(message)]`
+
+### Custom Types
+
+ink! supports custom type definitions needed to interact with the Substrate runtime. By default,
+ink! has access to all the [primitive Substrate types](conceptual/runtime/primitives.md) like
+`AccountId`, `Balance`, `Hash`, etc...
+
+You can specify and interact with additional types which may be specific to your Substrate runtime
+environment.
+
+### Overflow Safety
+
+Being written in Rust, ink! can provide compile-time overflow/underflow safety. Using a Rust
+compiler configuration, you can specify whether you want to support overflowing math, or if you want
+contract execution to panic when overflows occur. No need to continually import "Safe Math"
+libraries, although Rust also provides [integrated checked, wrapped, and saturated math
+functions](https://doc.rust-lang.org/std/primitive.u32.html).
```

Comment:
In the live-demo we have seen that this functionality currently is somewhat broken and we have to investigate further. Also note that we are currently thinking about techniques to limit the overflow-checks to user code which also might change how this whole thing works in general.

Reply:
I don't understand, can you elaborate? What is broken here?

Reply:
Denis found that it was not respecting the overflow option that Robin uses. It is overflowing and there are a few way to circumvent it. But up to now I think that there is no  final decision on how to proceed.